arch/risc-v/src/mpfs/mpfs_ihc.c

```diff
@@ -86,24 +86,23 @@
 #define VRINGS                   0x02        /* Number of vrings          */
 #define VRING_ALIGN              0x1000      /* Vring alignment           */
 #define VRING_NR                 256         /* Number of descriptors     */
-#define VRING_SIZE               512         /* Size of one descriptor    */
+#define VRING_SIZE               612         /* Size of one descriptor    */
```

Comment:
do the N descriptors need to align on the boundary of VRING_ALIGN? With this size it doesn't... I am not sure why the VRING_SIZE needs to be increased?

Reply:
This probably can be also 512, but we need to set smaller PKTSIZE for TUN interface.
I was confused because in CONFIG_NET_ETH_PKTSIZE it says "_IPv4 hosts are required to be able to handle an MSS of at least 536 octets, resulting in a minimum buffer size of 536+20+20+14 = 590._"

Reply:
ah, but this is right. RFC 791 states "Every internet destination must be able to receive a datagram of 576
    octets either in one piece or in fragments to be reassembled." The "14" in that equation is the ethernet header size, and we don't have ethernet header in TUN  interface, so

 as (if?) we don't support fragmenting, the descriptor needs to be able to hold IP packets of sizee 576 octets.

Reply:
Yes, I would like to avoid fragmentation. In our case descriptor need to be able to hold IP (576) + RPMSG net header (20) + RPMSG header (16).